lib/matplotlib/_mathtext_data.py

```diff
@@ -1110,198 +1110,632 @@
     '}'                      : 0x7d,
 }
 
-# Each element is a 4-tuple of the form:
-#   src_start, src_end, dst_font, dst_start
-#
-stix_virtual_fonts: dict[str, dict[str, list[tuple[int, int, str, int]]] |
```

Comment:
You are losing the type hint from this line, which appears to be the cause of the mypy errors.

Reply:
Passing it through the loop below for the ord check changes the type of the tuple to `Tuple(str, str, str, str)` which is different from the predefined `Tuple(int, int, str, int)` which causes `mypy` to throw an error. Any suggestion on how this can be handled better?

Reply:
Change the defined format to the new one?

Reply:
There are a couple of issues with just updating the type of the dictionary. The new format for `stix_virtual_fonts`-- updated to Unicode names-- is in type `tuple[str, str, str, (int | str)]` which is what we want to change it to, but the dictionary used in `StixFonts` requires the type `tuple[int, int, str, int]` mapping. The type of value changes during the course and `mypy` seems to complain about that, because it wants to keep the `type` of the dictionary intact.

I guess the above issues get fixed with the new changes, but `mypy` still seems to complain about indexing and assignment type errors... [🤷](https://emojipedia.org/person-shrugging)

Reply:
One option would be to use two different variables for each, though preferably one of them would be garbage collected pretty quick (maybe use `del` if it is top level, then)

Reply:
It looks like the assignment and indexing errors come from nesting the indexes like: `stix_virtual_fonts[keys][i]` where `mypy` seems to lose the type information and checks for `str` index in `dict`, but the `stix_virtual_fonts[keys][i]` type is a nested list with index:` int`. 

This problem seems related to some of the open issues reported in `mypy` GitHub repo:
https://github.com/python/mypy/issues/10146
https://github.com/python/mypy/issues/13374
https://github.com/python/mypy/issues/9176

A way would be to refactor the `stix_virtual_fonts table` to utilize `mypy's TypedDict` [option](https://mypy.readthedocs.io/en/stable/typed_dict.html), which requires a fixed schema to define the types, but that doesn't seem reasonable.

For lack of a better solution, I added the `# type: ignore` flag against the nested assignment and indexing, but that doesn't seem to work either... [🤦](https://emojipedia.org/person-facepalming)